cherab/solps/eirene/parser/fort44_post_2016.py

```diff
@@ -0,0 +1,135 @@
+
+        # Read Species labels
+        species_labels = []
+        for _ in range(ns):
+            line = file_handle.readline()
+            species_labels.append(line.strip())
+        if debug:
+            print("Species labels => {}".format(species_labels))
+
+        # Read all blocks of data from the file into a dictionary.
+        raw_data = {}
+        while True:
+            try:
+                name, values = read_labelled_block44(file_handle)
+            except EOFError:
+                break
+            else:
+                raw_data[name] = values
+
+        # create eirene object
+        eirene = Eirene(nx, ny, na, nm, ni, ns, species_labels, version)
+
+        # Read atomic species density and temperature (da, ta)
+        eirene.da = raw_data['dab2'].reshape(eirene.na, eirene.ny, eirene.nx)
+        eirene.ta = raw_data['tab2'].reshape(eirene.na, eirene.ny, eirene.nx)
+        if debug:
+            print('Atomic Neutral Density nD0: ', eirene.da[0, :, 0])
+            print('Atomic Neutral Temperature TD0: ', eirene.ta[0, :, 0])
+
+        # Read molecular species density and temperature (dm, tm)
+        eirene.dm = raw_data['dmb2'].reshape(eirene.nm, eirene.ny, eirene.nx)
+        eirene.tm = raw_data['tmb2'].reshape(eirene.nm, eirene.ny, eirene.nx)
+
+        # Read ion species density and temperature (di, ti)
+        eirene.di = raw_data['dib2'].reshape(eirene.ni, eirene.ny, eirene.nx)
+        eirene.ti = raw_data['tib2'].reshape(eirene.ni, eirene.ny, eirene.nx)
+
+        # Read atomic and molecular radial particle fluxes (rpa, rpm)
+        eirene.rpa = raw_data['rfluxa'].reshape(eirene.na, eirene.ny, eirene.nx)
+        eirene.rpm = raw_data['rfluxm'].reshape(eirene.nm, eirene.ny, eirene.nx)
+
+        # Read atomic and molecular poloidal particle fluxes (ppa, ppm)
+        eirene.ppa = raw_data['pfluxa'].reshape(eirene.na, eirene.ny, eirene.nx)
+        eirene.ppm = raw_data['pfluxm'].reshape(eirene.nm, eirene.ny, eirene.nx)
+
+        # Read radial energy flux (rea, rem)
+        eirene.rea = raw_data['refluxa'].reshape(eirene.na, eirene.ny, eirene.nx)
+        eirene.rem = raw_data['refluxm'].reshape(eirene.nm, eirene.ny, eirene.nx)
+
+        # Read poloidal energy flux (pea, pem)
+        eirene.pea = raw_data['pefluxa'].reshape(eirene.na, eirene.ny, eirene.nx)
+        eirene.pem = raw_data['pefluxm'].reshape(eirene.nm, eirene.ny, eirene.nx)
+
+        # Halpha total & molecules (emist, emism)
+        # fort.44 gives emission due to atoms and emission due to molecules separately.
+        eirene.emism = raw_data['emissmol'].reshape(1, eirene.ny, eirene.nx)
+        eirene.emist = raw_data['emiss'].reshape(1, eirene.ny, eirene.nx) + eirene.emism
+
+        # Radiated power (elosm, edism, eradt)
+        eirene.edism = raw_data['edissml'].reshape(eirene.nm, eirene.ny, eirene.nx)
+
+        eneutrad = raw_data['eneutrad'].reshape(eirene.na, eirene.ny, eirene.nx)
```

Comment:
Depending on SOLPS version and Eirene format, `eneutrad` may be resolved or summed over the atoms. So I propose changing this to:
```suggestion
        n_eneutrad = raw_data['eneutrad'].size // (eirene.ny * eirene.nx)
        eneutrad = raw_data['eneutrad'].reshape(n_eneutrad, eirene.ny, eirene.nx)
```

Reply:
OK, I'll modify it to deal with possibly being resolved over atoms too. I think it would be better to present a consistent interface in the `Eirene` object though: the shape of `Eirene.eradt` should not depend on whether the raw data is resolved or not. I propose we always provide `eradt` summed over all atoms, with shape `(1, ny, nx)`: this matches the existing behaviour with the original 2013 parser and is also consistent with the `emist` and `emism` data.

Reply:
@vsnever do you know which SOLPS/Eirene combinations result in eneutrad being summed over the atoms? If it's only going to be the case for pre-2016 (i.e. unlabelled) fort.44s then I think we can simplify this parser and move the check into the other parsers. Reducing the number of branches in each parser is a major advantage of using different parsers for different fort.44 versions.

Reply:
I'm afraid, this is more complicated than just a version of the fort.44:
```fortran
        if (jvft44 .ge. 20071209) then

          if (streql(solps_version,'4.3').or.streql(solps_version,'5.1').or.streql(eirene_format,'facelift')) then
C           eneutrad data is summed over species!
            call gfsub2(44,ndx,ndy,nnx,nny,eneutrad(:,:,1,0))
          else
C           eneutrad data is resolved over species!
            do is = 1, natmi
              call gfsub2(44,ndx,ndy,nnx,nny,eneutrad(:,:,is,0))
            end do
          end if

        end if
```
This is from b2yt_ngread.F. I don't know if a combination of post-2016 fort.44 and `solps_version` < 5.2 is possible or not, but I saw references to the latest (2019) fort.44 version even in the source code of SOLPS 4.3.3.